presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSpiller.java

```diff
@@ -93,7 +97,13 @@ public BinaryFileSpiller(
     private void writePages(Iterator<Page> pageIterator, Path spillPath)
     {
         try (SliceOutput output = new OutputStreamSliceOutput(new BufferedOutputStream(new FileOutputStream(spillPath.toFile())))) {
-            totalSpilledDataSize.addAndGet(PagesSerdeUtil.writePages(serde, output, pageIterator));
+            while (pageIterator.hasNext()) {
+                Page page = pageIterator.next();
+                long pageSize = page.getSizeInBytes();
+                localSpillContext.updateBytes(pageSize);
```

Comment:
I change it to serialize page in this method and account `SerializedPage.getPageSize` instead.